std/algorithm/iteration.d

```diff
@@ -11,11 +11,21 @@ $(T2 cache,
         Eagerly evaluates and caches another range's $(D front).)
 $(T2 cacheBidirectional,
         As above, but also provides $(D back) and $(D popBack).)
+$(T2 cartesianPower,
+        Lazily computes the Cartesian product of $(D r) for a number of
```

Comment:
I think the text is confusing, e.g. you want "Lazily computes the Cartesian product of `r` with itself..." The example should show a power of 3.

Also, I'd like more in terms of motivation. From the few people who need Cartesian products at all, a fraction would need the power of a range, and those could write 'cartesianProduct(r, r, r)`. True, that can't be done if the exponent is a variable. But how often does that happen?